src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,138 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private int count;
+    public static final int MAX_CAPACITY = 10;
+    private Object[] elements = new Object[MAX_CAPACITY];
     @Override
     public void add(T value) {
+        if (count >= elements.length) {
+            elements = grow();
+        }
+        elements[count] = value;
+        count++;
 
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexforAdd(index);
+        count++;
+        Object[] newArray = new Object[MAX_CAPACITY];
+        if (count > MAX_CAPACITY) {
+             newArray = grow();
+
+        }
+        for (int i = 0; i < newArray.length; i++) {
+            if (i == index) {
+                break;
+            }
+            newArray[i] = elements[i];
+        }
+        newArray[index] = value;
+        for (int i = index + 1; i < newArray.length; i++) {
+            newArray[i] = elements[i - 1];
+        }
+        elements = newArray;
+        elements[index] = value;
```

Comment:
This line is redundant because `newArray[index] = value;` has already been set. It should be removed.